gene2phenotype_project/gene2phenotype_app/models.py

```diff
@@ -175,24 +174,53 @@ class LGDVariantGenccConsequence(models.Model):
     lgd = models.ForeignKey("LocusGenotypeDisease", on_delete=models.PROTECT)
     variant_consequence = models.ForeignKey("OntologyTerm", on_delete=models.PROTECT)
     support = models.ForeignKey("Attrib", on_delete=models.PROTECT)
-    panel = models.ForeignKey("Panel", on_delete=models.PROTECT, null=True, default=None)
-    publication = models.ForeignKey("Publication", on_delete=models.PROTECT, null=True, default=None)
     is_deleted = models.SmallIntegerField(null=False, default=False)
     history = HistoricalRecords()
 
     class Meta:
         db_table = "lgd_variant_gencc_consequence"
-        unique_together = ["lgd", "variant_consequence", "support", "publication"]
+        unique_together = ["lgd", "variant_consequence", "support"]
+
+class CVMolecularMechanism(models.Model):
+    """
+        Controlled vocabulary for molecular mechanism
+    """
+
+    # Type of vocabulary available for the mechanism
+    choices_values = (
+        ("mechanism", "Molecular mechanism"),
+        ("mechanism_synopsis", "Mechanism synopsis"),
+        ("support", "Mechanism support"),
+        ("evidence", "Mechanism evidence")
+    )
+
+    # Type of vocabulary available for the evidence
+    # The evidence values are a subtype of type "evidence"
+    choices_evidence_types = (
+        ("function", "Function"),
+        ("rescue", "Rescue"),
+        ("functional_alteration", "Functional alteration"),
+        ("models", "Models")
+    )
+
+    id = models.AutoField(primary_key=True)
+    type = models.CharField(max_length=50, choices=choices_values)
+    # the subtype is only populated for the evidence
+    subtype = models.CharField(max_length=100, choices=choices_evidence_types, null=True)
+    value = models.CharField(max_length=100)
+
+    class Meta:
+        db_table = "cv_molecular_mechanism"
+        unique_together = ["type", "subtype", "value"]
 
 class LGDMolecularMechanism(models.Model):
     id = models.AutoField(primary_key=True)
     lgd = models.ForeignKey("LocusGenotypeDisease", on_delete=models.PROTECT)
-    mechanism = models.ForeignKey("Attrib", related_name='mechanism', on_delete=models.PROTECT)
-    mechanism_support = models.ForeignKey("Attrib", related_name='mechanism_support', on_delete=models.PROTECT)
-    synopsis = models.ForeignKey("Attrib", related_name='synopsis', on_delete=models.PROTECT, null=True)
-    synopsis_support = models.ForeignKey("Attrib", related_name='synopsis_support', on_delete=models.PROTECT, null=True)
+    mechanism = models.ForeignKey("CVMolecularMechanism", related_name='mechanism', on_delete=models.PROTECT)
```

Comment:
Also suggesting we add that this can not be null

Reply:
By default, it doesn't allow null values.
This is how this model looks in the db:
![Screenshot 2024-06-21 at 16 51 30](https://github.com/EBI-G2P/gene2phenotype_api/assets/44366649/4a8a5513-a28e-4ed4-9e07-8cd302e22303)

Reply:
Okay, just might be better for it to reflect it in the code.  Do not know if you do not think thats a good idea

Reply:
I'll add the `null=False` to the next PR.